src/manager.rs

```diff
@@ -120,31 +120,38 @@ impl<S: MutEventSubscriber> EventManager<S> {
     }
 
     fn dispatch_events(&mut self, event_count: usize) {
+        // Used to record whether there's an endpoint event that needs to be handled.
+        #[cfg(feature = "remote_endpoint")]
+        let mut endpoint_event = None;
+
         // Use the temporary, pre-allocated buffer to check ready events.
         for ev_index in 0..event_count {
             let event = self.ready_events[ev_index];
+            let fd = event.fd();
 
-            #[cfg(feature = "remote_endpoint")]
-            {
-                if self.dispatch_endpoint_event(event) {
-                    continue;
+            if let Some(subscriber_id) = self.epoll_context.subscriber_id(fd) {
+                self.subscribers.get_mut_unchecked(subscriber_id).process(
+                    Events::with_inner(event),
+                    // The `subscriber_id` is valid because we checked it before.
+                    &mut self.epoll_context.ops_unchecked(subscriber_id),
+                );
+            } else {
+                #[cfg(feature = "remote_endpoint")]
+                {
+                    if fd == self.channel.fd() {
+                        endpoint_event = Some(event);
+                        continue;
+                    }
                 }
-            }
 
-            let fd = event.fd();
-            // This error condition can happen when the fd associated with a subscriber was closed,
-            // and the subscriber did not handle the RHUP/
-            let subscriber_id = self
-                .epoll_context
-                .subscriber_id(fd)
-                .expect("Received event on fd from subscriber that does not exist");
-
-            self.subscribers.get_mut_unchecked(subscriber_id).process(
-                Events::with_inner(event),
-                // The `subscriber_id` is valid because we checked it before.
-                &mut self.epoll_context.ops_unchecked(subscriber_id),
-            );
+                // This error condition can happen when the fd associated with a subscriber was
+                // closed, and the subscriber did not handle the RHUP.
+                panic!("Received event on fd from subscriber that does not exist");
```

Comment:
I am not particularly happy with this way of handling the error, since nothing is actually forcing the user to handle RHUP (or other errors for that matter). Should we log an issue with this and try to address it in the next release?

Reply:
On second though, we shouldn't really reach this situation during normal operation. I've dropped the additional commit from the PR (it's still available in [this](https://github.com/alexandruag/event-manager/tree/temp2) branch), and switched to an `unreachable` statement. Let's sync and see if we're aligned on this.